test/extended/networking/egress_firewall.go

```diff
@@ -100,6 +125,44 @@ func doEgressFwTest(f *e2e.Framework, oc *exutil.CLI, manifest string, nodeSelec
 
 	o.Expect(sendEgressFwTraffic(f, oc, egressFWTestPod, nodeSelectorSupport)).To(o.Succeed())
 
+	// TODO:
+	// - Check for coredns-ocp-dnsnameresolver plugin.
+	// - Remove TecHPreview check when feature is GA.
+	if networkPluginName() == OVNKubernetesPluginName && exutil.IsTechPreviewNoUpgrade(oc) {
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err := dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dnsNameResolverClient := dc.Resource(schema.GroupVersionResource{Group: "network.openshift.io", Resource: "dnsnameresolvers", Version: "v1alpha1"})
+
+		e2e.Logf("fetching DNSNameResolver objects list")
+		obj, err := dnsNameResolverClient.List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		dnsNameResolvers := objx.Map(obj.UnstructuredContent())
+		items := objects(dnsNameResolvers.Get("items"))
+		o.Expect(len(items)).ShouldNot(o.Equal(0))
+
+		for _, item := range items {
+			dnsName := item.Get("spec.name").String()
+			resolvedNames := item.Get("status.resolvedNames").InterSlice()
+			o.Expect(len(resolvedNames)).Should(o.Equal(1))
+			name, ok := resolvedNames[0].(nwv1alpha1.DNSNameResolverResolvedName)
+			o.Expect(ok).Should(o.Equal(true))
+			switch dnsName {
+			case "docs.openshift.com.":
+				o.Expect(name.DNSName).Should(o.Equal("docs.openshift.com."))
+				o.Expect(name.ResolutionFailures).Should(o.Equal(0))
+			case "*.google.com.":
+				o.Expect(name.DNSName).Should(o.Equal("www.google.com."))
+				o.Expect(name.ResolutionFailures).Should(o.Equal(0))
+			default:
+				e2e.Logf("DNSNameResolver not expected to be created for DNSName %s", dnsName)
+				o.Expect(dnsName).NotTo(o.HaveOccurred())
+			}
```

Comment:
do we even need this part? considering it's a part of egress firewall test, we only care about connectivity allowed/denied, I would say we should have a separate test suite for dnsNameResolver, where we would check if the status is correct (potentially later)

Reply:
Since it was a new feature added, thought to verify it as well. Should I remove it or keep it so that when this block code is moved to dedicated test suite, this would be moved as well?

Reply:
I would say it makes sense to remove it for now until we have a separate test set for dnsNameResolver
@arkadeepsen do you agree?

Reply:
I agree. We can plan for adding it in a separate test when we plan for GA release of the feature. For now removing it makes sense.

Reply:
suggestion incorporated.